src/mpid/ch4/shm/src/shm_p2p.h

```diff
@@ -160,7 +162,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_SHM_mpi_isend(const void *buf, MPI_Aint count
 
     MPIDI_Datatype_check_contig_size(datatype, count, dt_contig, data_sz);
     if (MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE > -1 &&
-        dt_contig && data_sz > MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE) {
+        dt_contig && data_sz > MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE && rank != MPIR_Process.rank) {
```

Comment:
We need to add `rank != MPIR_Process.rank` condition check in a separate commit. It is not specific for coop-copy.